packages/matter.js/src/behavior/definitions/operational-credentials/DeviceCertification.ts

```diff
@@ -69,4 +117,6 @@ export namespace DeviceCertification {
         intermediateCertificate: ByteArray;
         declaration: ByteArray;
     }
+
+    export type ProviderFunction = () => Promise<DeviceCertification.Configuration>;
```

Comment:
Seems like the type we mostly use is config | provider so maybe could define as e.g.:

```js
    export type Definition = Configuration | (() => Promise<Configuration>)
```

...then the union would only appear once